src/main.rs

```diff
@@ -60,7 +60,8 @@ fn main() {
             "target/release"
         });
 
-        let libconfig_path = so_dir.join("libconfig.so");
+        let libconfig_str = resolve_config_file();
+        let libconfig_path = so_dir.join(libconfig_str);
```

Comment:
Why not passing it directly?

```suggestion
        let libconfig_path = so_dir.join(resolve_config_file());
```